source/uwp/Renderer/lib/AdaptiveShowCardActionRenderer.h

```diff
@@ -26,6 +26,12 @@ namespace AdaptiveNamespace
                                 _In_ ABI::AdaptiveNamespace::IAdaptiveActionParserRegistration* actionParserRegistration,
                                 _In_ ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveNamespace::AdaptiveWarning*>* adaptiveWarnings,
                                 _COM_Outptr_ ABI::AdaptiveNamespace::IAdaptiveActionElement** element) noexcept override;
+
+        static HRESULT BuildShowCard(_In_ ABI::AdaptiveNamespace::IAdaptiveCard* showCard,
+                                     _In_ ABI::AdaptiveNamespace::IAdaptiveRenderContext* renderContext,
+                                     _In_ ABI::AdaptiveNamespace::IAdaptiveRenderArgs* renderArgs,
+                                     bool isBottomActionBar,
```

Comment:
>bool [](start = 37, length = 4)

should we try to consolidate bool, boolean, BOOL and BOOLEAN? I understand some types are required by some APIs but we have 4 ways of representing true and false

Reply:
this has been tricky for a while. I think the basics are:

* whenever possible, use `bool`
* WinRT contracts require `boolean` (not positive, but guessing this is to deal with the cross-language nature of WinRT)
* BOOL and BOOLEAN are holdovers from Win32 (and Win16 before it), which originated in a time before `bool` was a real type in C. they should only be used as required to interact with non-WinRT Windows APIs. 

of course, like any style guidance, there's room for interpretation. you probably wouldn't introduce the only use of `bool` in a file that uses `BOOL` everywhere (unless, I guess, you can confidently switch everything to `bool`)

---
In reply to: [326399601](https://github.com/microsoft/AdaptiveCards/pull/3457#discussion_r326399601) [](ancestors = 326399601)